great_expectations/core/datasource_dict.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import copy
+import logging
+from collections import UserDict
+from typing import TYPE_CHECKING
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations.data_context.types.base import (
+    datasourceConfigSchema,
+)
+from great_expectations.datasource.new_datasource import BaseDatasource
+
+if TYPE_CHECKING:
+    from great_expectations.core.config_provider import _ConfigurationProvider
+    from great_expectations.data_context.data_context.abstract_data_context import (
+        AbstractDataContext,
+    )
+    from great_expectations.data_context.store.datasource_store import DatasourceStore
+    from great_expectations.datasource.fluent import Datasource as FluentDatasource
+
+logger = logging.getLogger(__name__)
+
+
+class DatasourceDict(UserDict):
```

Comment:
Currently only used in Cloud but I'd like to expand to the base class if I have time - for now, leaving it alone